pkg/codegen/dotnet/gen.go

```diff
@@ -2158,11 +2158,19 @@ func genPackageMetadata(pkg *schema.Package,
 	if err != nil {
 		return err
 	}
-	plugin, err := (&plugin.PulumiPluginJSON{
+
+	pulumiPlugin := &plugin.PulumiPluginJSON{
 		Resource: true,
 		Name:     pkg.Name,
 		Server:   pkg.PluginDownloadURL,
-	}).JSON()
+	}
+
+	if pkg.Version != nil && codegen.RespectVersion() {
+		files.add("version.txt", []byte(pkg.Version.String()))
```

Comment:
What is version.txt supposed to be? If we're stamping a version to the codegen'd project should it be stamped into the project.csproj as <AssemblyVersion>?

Reply:
`version.txt` is the file we use to check the version of the package (at runtime). It makes some amount of sense to stamp the version into `*.csproj`.